sheepdog/transactions/deletion/entity.py

```diff
@@ -95,24 +95,20 @@ def _delete(self):
     def recursive_test_deletion(self):
         self.logger.info('Attempting deletion of {}'.format(self))
 
-        subentities = [
-            DeletionEntity(self.transaction, n)
-            for n in self.neighbors if n
-        ]
-
-        self.transaction.graph_validator.record_errors(
-            self.transaction.db_driver, subentities)
-
-        invalid_neighbor_entities = [e for e in subentities if not e.is_valid]
-
-        for entity in invalid_neighbor_entities:
-            if entity.node_exists:
-                self.dependents[entity.node.node_id] = entity
-                entity._delete()
-                entity.recursive_test_deletion()
-
-                # Add child dependencies to this entity
-                self.dependents.update(entity.dependents)
+        for n in self.neighbors:
+            if n:
+                subentity = DeletionEntity(self.transaction, n)
+                self.transaction.graph_validator.record_errors(
+                    self.transaction.db_driver, [subentity])
+                if not subentity.is_valid and subentity.node_exists:
+                    self.dependents[subentity.node.node_id] = subentity
+                    subentity._delete()
+                    subentity.recursive_test_deletion()
+
+                    # Add child dependencies to this entity
+                    self.dependents.update(subentity.dependents)
+            if self.dependents:
```

Comment:
should put a comment saying that we only get the first dependent for performance. also we should change the `self.neighbors` assignment to be a generator that lazily yield neibhors, otherwise at the class initiation it still loops through alll edges

Reply:
https://github.com/uc-cdis/sheepdog/blob/aa6cd292e220ee217af146e86593cd411baa1138/sheepdog/transactions/deletion/entity.py#L87 the actual deletion should not get all edges too, since postgres take care of cascading to edge deletion